code/modules/reagents/chemistry/holder.dm

```diff
@@ -163,10 +163,12 @@
  * * ignore splitting - Don't call the process that handles reagent spliting in a mob (impure/inverse) - generally leave this false unless you care about REAGENTS_DONOTSPLIT flags (see reagent defines)
  */
 /datum/reagents/proc/add_reagent(reagent, amount, list/data=null, reagtemp = DEFAULT_REAGENT_TEMPERATURE, added_purity = null, added_ph, no_react = FALSE, override_base_ph = FALSE, ignore_splitting = FALSE)
-	if(!isnum(amount) || !amount)
+	// Prevents small amount problems, as well as zero and below zero amounts.
+	if(amount <= CHEMICAL_QUANTISATION_LEVEL)
```

Comment:
I think moving this forward of the other check just to get unit tests to pass is honestly a mistake.

Getting sent 0 amounts is still an issue that deserves investigating in upstream codes

Reply:
I thought the same, but it's a lot of effort, nobody checks the amount when function is capable of silently ignoring them.

Reply:
But if you think this is necessary, I won't interfere.